nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/JacksonCSVRecordReader.java

```diff
@@ -108,6 +112,17 @@ public Record nextRecord(final boolean coerceTypes, final boolean dropUnknownFie
                     rawFieldNames = schema.getFieldNames();
                 } else {
                     rawFieldNames = Arrays.asList(csvRecord);
+                    if (rawFieldNames.size() > schema.getFieldCount() && !allowDuplicateHeaderNames) {
+                        final Set<String> deDupe = new HashSet<>(schema.getFieldCount());
```

Comment:
So, if I have multiple duplicate names, then I'm going to have to iterate through these errors, one by one, field by field.
Have you given thought to tracking the duplicates, and then throwing if there are any, and include all the duplicate fields in that exception?

Reply:
Yes. But that's because I've replicated the same functionality as Apache Commons CSV, which reports a single error (somewhat annoyingly).

We could indeed have a separate approach for Jackson (which I actually coded first), but I decided to keep the experience as close as possible irrespective of chosen Parser.

Happy to discuss/be overruled on that.

Reply:
That is fair enough